liquid.js

```diff
@@ -12,17 +12,52 @@ engine.registerFileSystem(
 // https://github.com/npm/node-semver#coercion
 
 Liquid.Condition.operators.ver_gt = (cond, left, right) => {
-  if (!startsWithNumber(left)) return false
-  return semver.gt(semver.coerce(left), semver.coerce(right))
+  if (!matchesVersionString(left)) return false
+  if (!matchesVersionString(right)) return false
+
+  const [leftPlan, leftRelease] = splitVersion(left)
+  const [rightPlan, rightRelease] = splitVersion(right)
+
+  if (leftPlan !== rightPlan) return false
+
+  return semver.gt(semver.coerce(leftRelease), semver.coerce(rightRelease))
 }
 
 Liquid.Condition.operators.ver_lt = (cond, left, right) => {
-  if (!startsWithNumber(left)) return false
-  return semver.lt(semver.coerce(left), semver.coerce(right))
+  if (!matchesVersionString(left)) return false
+  if (!matchesVersionString(right)) return false
+
+  const [leftPlan, leftRelease] = splitVersion(left)
+  const [rightPlan, rightRelease] = splitVersion(right)
+
+  if (leftPlan !== rightPlan) return false
+
+  return semver.lt(semver.coerce(leftRelease), semver.coerce(rightRelease))
 }
 
 module.exports = engine
 
-function startsWithNumber (input) {
-  return input && input.match(/^\d+/)
+function matchesVersionString (input) {
+  return input && input.match(/(?:^[a-z](?:[a-z-]*[a-z])?@)?\d+(?:\.\d+)*/)
```

Comment:
Actually we should move the `^` to outside of the optional grouping:

```suggestion
  return input && input.match(/^(?:[a-z](?:[a-z-]*[a-z])?@)?\d+(?:\.\d+)*/)
```